doc/lib/mqtt.txt

```diff
@@ -0,0 +1,60 @@
+/**
+@mainpage
+@anchor mqtt
+@brief MQTT 3.1.1 client library.
+
+MQTT is the protocol that allows a device to communicate with AWS IoT. The MQTT library provided with this SDK implements a subset of the MQTT v3.1.1 protocol. The sample application for Linux `subscribe_publish_sample.c` is provided as a starting point for using this library.
```

Comment:
Same concern here about the sentence being misleading about MQTT only being an AWS IoT protocol. I suggest the similar modification of "MQTT being one of the protocols that AWS IoT supports."

Reply:
Fixed